src/commonMain/kotlin/exchange.dydx.abacus/output/Account.kt

```diff
@@ -1488,6 +1555,210 @@ data class TradingRewards(
+                                val synced = HistoricalTradingReward.create(null, parser, item)
+                                addHistoricalTradingRewards(result, synced!!, period, lastStart)
+                                result.add(synced)
+                                dataIndex++
+                                lastStart = synced.startedAtInMilliseconds
+                            }
+
+                            (comparison == ComparisonOrder.descending) -> {
+                                // item is older than obj
+                                addHistoricalTradingRewards(result, obj, period, lastStart)
+                                result.add(obj)
+                                objIndex++
+                                lastStart = obj.startedAtInMilliseconds
+                            }
+
+                            else -> {
+                                val synced = HistoricalTradingReward.create(obj, parser, item)
+                                addHistoricalTradingRewards(result, obj, period, lastStart)
+                                result.add(synced!!)
+                                objIndex++
+                                dataIndex++
+                                lastStart = synced.startedAtInMilliseconds
+                            }
+                        }
+                    } else {
+                        dataIndex++
+                    }
+                }
+                while (objIndex < objs!!.size) {
+                    val obj = objs[objIndex]
+                    addHistoricalTradingRewards(result, obj, period, lastStart)
+                    result.add(obj)
+                    objIndex++
+                    lastStart = obj.startedAtInMilliseconds
+                }
+                while (dataIndex < data.size) {
+                    val item = parser.asMap(data[dataIndex])
+                    val itemStart =
+                        parser.asDatetime(item?.get("startedAt"))?.toEpochMilliseconds()?.toDouble()
+                    if (item != null && itemStart != null) {
+                        val synced = HistoricalTradingReward.create(null, parser, item)
+                        addHistoricalTradingRewards(result, synced!!, period, lastStart)
+                        result.add(synced)
+                        dataIndex++
+                        lastStart = synced.startedAtInMilliseconds
+                    }
+                }
+            } else {
+                result.add(currentPeriodPlaceHolder(period))
+            }
+            return result
+        }
+
+        private fun currentPeriodPlaceHolder(period: String): HistoricalTradingReward {
+            val now = Clock.System.now()
+            val thisPeriod = when (period) {
+                "DAILY" -> today(now)
+                "WEEKLY" -> thisWeek(now)
+                "MONTHLY" -> thisMonth(now)
+                else -> today(now)
+            }
```

Comment:
to clarify, shouldn't we always know the current period's start time from the api response? (i.e. the most recent period should be the current period) so we should only need to calculate the end time?